src/S3/Exception/S3MultipartUploadException.php

```diff
@@ -0,0 +1,86 @@
+<?php
+namespace Aws\S3\Exception;
+
+use Aws\CommandInterface;
+use Aws\Exception\AwsException;
+use Aws\Multipart\UploadState;
+
+class S3MultipartUploadException extends \Aws\Exception\MultipartUploadException
+{
+    /** @var string Bucket of the transfer object */
+    private $bucket;
+    /** @var string Key of the transfer object */
+    private $key;
+    /** @var string Source file name of the transfer object */
+    private $filename;
+
+    /**
+     * @param UploadState      $state Upload state at time of the exception.
+     * @param \Exception|array $prev  Exception being thrown. Could be an array of
+     *                                AwsExceptions being thrown when uploading parts
+     *                                for one object, or an instance of AwsException
+     *                                for a specific Multipart error being thrown in
+     *                                the MultipartUpload process.
+     */
+    public function __construct(UploadState $state, $prev = null) {
+        if (is_array($prev)) {
+            foreach ($prev as $part => $error) {
+                $this->collectPathInfo($error->getCommand());
```

Comment:
`$this->collectPathInfo` overwrites the `$bucket`, `$key`, and `$filename` each time it is called. Since exceptions encountered when performing a multipart upload would always pertain to the same bucket and key, I don't think this is a bug so much as it is an inefficiency.

Would it make sense just have a compound condition for the if statement that ensures the array is not empty and collects path info from either the head or the tail of that array, whichever represents the most recently encountered error? E.g.:

```php
if (is_array($prev) && $error = array_pop($prev) {
    $this->collectPathInfo($error->getCommand);
}
```